### PR TITLE
[REF] stock_location_code: Use new api _name_get method

### PR DESCRIPTION
**Pull request by moylop260**
Dummy PR: https://github.com/Vauxoo/yoytec/issues/1210
# Without this PR

The main reason of this PR is reduce the cProfile performance.

Without this changes we have the following cProfile output in runbot:

``` bash
Print cProfile sorted by  cumulative
ncalls     tottime    tt_percall cumtime    ct_percall file:lineno (method)
16723      0.092      0.000      1.438      0.000      addons-vauxoo/stock_location_code/model/stock.py:81 (_name_get)
4532       0.040      0.000      0.621      0.000      addons-vauxoo/website_product_availability/models/product.py:151 (_product_availability_warehouse)
```

screen shot 2016-05-13 at 3 28 41 pm
# With this PR

The cProfile output of this PR was:
screen shot 2016-05-13 at 4 01 08 pm

``` bash
Print cProfile sorted by  cumulative
ncalls     tottime    tt_percall cumtime    ct_percall file:lineno (method)
2997       0.046      0.000      0.633      0.000      /root/odoo-themes/website_less/website.py:18 (_image)
4610       0.040      0.000      0.563      0.000      /root/addons-vauxoo/website_product_availability/models/product.py:151 (_product_availability_warehouse)
4566       0.007      0.000      0.400      0.000      /root/addons-vauxoo/partner_foreign/model/res_partner.py:34 (_get_location_options)
1083       0.005      0.000      0.286      0.000      /root/partner-contact/base_partner_sequence/models/partner.py:52 (write)
5717       0.009      0.000      0.281      0.000      /root/addons-yelizariev/web_debranding/models/ir_translation.py:23 (_get_source)
2941       0.015      0.000      0.270      0.000      /root/build/Vauxoo/yoytec/yoytec/model/website.py:17 (sale_get_order)
4207       0.016      0.000      0.252      0.000      /root/build/Vauxoo/yoytec/partner_long_name/model/res_partner.py:19 (name_get)
1427       0.003      0.000      0.141      0.000      /root/build/Vauxoo/yoytec/yoytec_crm_claim_partner_foreign/model/crm_claim.py:33 (_get_related_location_options)
1553       0.011      0.000      0.139      0.000      /root/build/Vauxoo/yoytec/yoytec_partner_credit_limit/model/partner.py:68 (_get_credit_overloaded)
1609       0.012      0.000      0.127      0.000      /root/build/Vauxoo/yoytec/yoytec_sale_workflow/models/sale_order.py:147 (compute_allowed_credit_sale)
1075       0.005      0.000      0.114      0.000      /root/build/Vauxoo/yoytec/yoytec_checkout/models/website_default_fields.py:89 (show_required_asterisk)
4629       0.013      0.000      0.093      0.000      /root/build/Vauxoo/yoytec/yoytec_delivery/models/delivery_grid.py:106 (grid_get)
1548       0.007      0.000      0.086      0.000      /root/build/Vauxoo/yoytec/yoytec_partner_credit_limit/model/partner.py:138 (get_allowed_sale)
689        0.005      0.000      0.080      0.000      /root/addons-vauxoo/stock_allow_past_date/model/stock_move.py:25 (has_validate_picking_after_move_date)
1438       0.005      0.000      0.078      0.000      /root/stock-logistics-workflow/product_unique_serial/model/stock.py:101 (_check_inicity_lot_product)
1194       0.005      0.000      0.076      0.000      /root/build/Vauxoo/yoytec/website_yoytec/models/website.py:33 (get_public_user_id)
1086       0.006      0.000      0.074      0.000      /root/build/Vauxoo/yoytec/yoytec_partner_credit_limit/model/partner.py:92 (_get_overdue_credit)
282        0.004      0.000      0.070      0.000      /root/odoo-themes/website_less/ir_attachment.py:7 (invalidate_bundle)
894        0.004      0.000      0.069      0.000      /root/addons-vauxoo/stock_allow_past_date/model/stock_move.py:37 (action_done)
668        0.003      0.000      0.062      0.000      /root/build/Vauxoo/yoytec/yoytec_delivery/models/res_partner.py:17 (_get_delivery_method)
213        0.002      0.000      0.054      0.000      /root/build/Vauxoo/yoytec/yoytec_checkout/models/sale_order.py:59 (_get_warehouses_availavility)
223        0.002      0.000      0.047      0.000      /root/build/Vauxoo/yoytec/yoytec_checkout/controllers/main.py:479 (onepage_confirm_address)
464        0.002      0.000      0.046      0.000      /root/odoo-panama/l10n_pa_localization/model/res_partner.py:192 (_display_address)
651        0.001      0.000      0.041      0.000      /root/stock-logistics-workflow/stock_no_negative/model/stock.py:32 (get_operations_as_action_done)
750        0.002      0.000      0.039      0.000      /root/build/Vauxoo/yoytec/sale_cash_register/models/sale_cash_register.py:503 (create)
5717       0.008      0.000      0.038      0.000      /root/addons-yelizariev/web_debranding/models/ir_translation.py:12 (_debrand)
471        0.004      0.000      0.038      0.000      /root/odoo-panama/l10n_pa_vat/model/res_partner.py:44 (check_vat_pa)
371        0.003      0.000      0.037      0.000      /root/addons-vauxoo/website_product_availability/models/product.py:131 (_get_planned_dates)
315        0.002      0.000      0.037      0.000      /root/stock-logistics-workflow/stock_no_negative/model/stock.py:66 (check_before_action_done)
591        0.002      0.000      0.037      0.000      /root/build/Vauxoo/yoytec/sale_cash_register/models/sale_cash_register.py:356 (_get_outstanding_info_JSON)
```

quilarque

Use default location name `{parent_location.name} / {location.name}`

### CODE REVIEW

**Comment by moylop260** [comment]
@nhomar @gquilarque 
Feeback here please
The performance is increased considerably with this change in cProfile reports.

**Comment by nhomar** [comment]
moylop?

this has unittest? can you help us adding it please it should be only 1 method.

**Comment by moylop260** [comment]
This method have a 100% of coverage
It's the reason that cProfile show a big time consuming in this method, because other models test it

Maybe, we can add all test cases in other PR because we need to know if this change don't affect the C.A. of the H.U. with @gquilarque

**Comment by moylop260** [comment]
@gquilarque 
The C.A. told us that the customer needs identify the warehouse because before you had a value with `stock` instead of `ED/stock`.

This should be fixed with a parent location:
`stock` with parent_id: `ED` to show `ED/stock` with default odoo way.
Now, we have a parent location and using the default odoo way we have `ED/stock`

Karen added warehouse name and removed the parent location name, then we have (before of this PR):
`stock ED`

Now, with this PR I added the default odoo way and the Karen way:
`ED/stock ED`
But this is redundant then the question is:
Could we let the default odoo way and remove the Karen way?

| Case | Result | Tech |
| --- | --- | --- |
| Odoo default way | ED/Stock | `location.parent_id.name / location.name` |
| Karen Way | Stock (ED) | `location.name (search_warehouse().name)` |
| Moy Way | ED/Stock (ED) | `location.parent_id.name / location.name (search_warehouse().name)` |

IMHO the best way is the odoo default way:
`ED/Stock` using `location.parent_id.name / location.name`
To avoid redundant info and let code in the default way without changes.

**Comment by gquilarque** [comment]
Hi @moylop260 

I vote for the 1st option because is the Odoo default way.

However I'll make a POC with this changes in sale.order and stock.picking for I can see if this change influence in the main reason that development initially requested as CA#5 HU#962.

**CA#5**. Que cuando las locations sean llamadas desde un campo m2o se muestren de la siguiente forma: 
-Stock (El Dorado) 
-Stock (San Francisco) 
Actualmente se muestra solo Stock.

**Comment by gquilarque** [comment]
@moylop260 

I did the POC and this change works properly as initially required.

POC:
1.- Create the sale.order:
Review:
- `sale_order.warehouse_id`: "El Dorado" [correct name]
- `stock_warehouse.wh_pack_stock_loc_id` and others..: "ED/Packing Zone (El Dorado) "  [correct name]

2.- Confirm sale.order.
3.- View stock.picking
Review:
- `stock._move.location_dest_id`: ED/Stock (El Dorado) [correct name]
- `stock_picking_type.default_location_dest_id`:  ED/Stock (El Dorado) [correct name]

**In conclusion:**
We can change this feature by the proposed @moy. I vote for the 1st option because is the Odoo default way and because the actual solution is redundant.

**Functional Test on:** http://10563-1210-ed5394.runbot2.vauxoo.com/?db=10563-1210-ed5394-all

**Comment by moylop260** [comment]
@gquilarque 
Thanks now I changed to option 1

@nhomar 
Now I added the tests cases

Thanks!

**Comment by moylop260** [comment]
Please, if It's merged don't squash to get a easiest way to revert and get full git log of option 3
